sdk/core/azure-core/src/main/java/com/azure/core/http/rest/PagedFluxBase.java

```diff
@@ -70,34 +67,38 @@ public PagedFluxBase(Supplier<Mono<P>> firstPageRetriever) {
      */
     public PagedFluxBase(Supplier<Mono<P>> firstPageRetriever,
                          Function<String, Mono<P>> nextPageRetriever) {
-        Objects.requireNonNull(firstPageRetriever, "'firstPageRetriever' cannot be null.");
-        Objects.requireNonNull(nextPageRetriever, "'nextPageRetriever' function cannot be null.");
-        this.firstPageRetriever = firstPageRetriever;
-        this.nextPageRetriever = nextPageRetriever;
-    }
-
-    Supplier<Mono<P>> getFirstPageRetriever() {
-        return firstPageRetriever;
+        this(() -> (continuationToken, pageSize) -> continuationToken == null
+            ? firstPageRetriever.get().flux()
+            : nextPageRetriever.apply(continuationToken).flux(), true);
     }
 
-    Function<String, Mono<P>> getNextPageRetriever() {
-        return nextPageRetriever;
+    /**
+     * PACKAGE INTERNAL CONSTRUCTOR, exists only to support the PRIVATE PagedFlux.ctr(Supplier, boolean)
+     * use case.
+     *
+     * Create PagedFlux backed by Page Retriever Function Supplier.
+     *
+     * @param provider the Page Retrieval Provider
+     * @param ignored ignored
+     */
+    public PagedFluxBase(Supplier<PageRetriever<String, P>> provider, boolean ignored) {
```

Comment:
Looks like you intended to make this a package-private ctr but it's marked as `public`.